Services/APIVersioning.md

```diff
@@ -0,0 +1,144 @@
+
+# Table of Contents
+
+* [Goals](#goals)
+* [API Versioning](#api-versioning)
+* [Responsibilities of Third-Party Clients](#responsibilities-of-third-party-clients)
+* [Responsibilities of Wikia-Authored Clients](#responsibilities-of-wikia-authored-clients)
+* [Responsibilities of API Developers](#responsibiltiies-of-api-developers)
+* [A Note on HATEOAS](#a-note-on-hateoas)
+
+# Goals
+
+We want to ensure the following:
+
+* Services can be updated independently, at any time
+* Multiple versions of the same service can run side-by-side (during upgrades, downgrades or during A/B testing)
+* API Clients are easy to develop and maintain
+
+We want to avoid the following:
+
+* Production outages due to API changes
+* Dependencies between upgrades
+* Breaking external clients, both official and unofficial
+
+# API Versioning
+
+APIs should not put any version information in their URI. The only exception to this rule is if the API is a transparent proxy for a third-party API that already does not follow this rule.
+
+Placing the version number in the API encourages backwards-incompatible changes through the draconic measure of copying the entire API, _including all the resources that weren't changed_ to a new location. This in turn creates the additional development, testing and support burden of having to maintain independently a stack of redundant APIs. Eventually the decision is made to limit maintenance costs by sunsetting old versions, and clients that would never have broken because the APIs they relied on were not the ones that caused the version bump, cease to work for no good reason.
+
+This complexity only increases when an API is composed from a number of cooperating services, and we would have to choose between clients having to track independent versions both of multiple sub-APIs (and the versions of dependencies between those APIs), or us having to revise the global version of _every_ API whenever an individual service changes.
+
+(All of this is also applies to performing version negotiation in HTTP headers.)
+
+A much better approach is to do your best to ensure APIs are backwards compatible, and if it becomes necessary necessary, create new resources at new locations only for those things that actually changed.
+
+# Responsibilities of Third-Party Clients
+
+Many of the clients for our APIs will be developed outside our control. We define a reasonable minimum set of expectations for third-party clients to place an upper bound on the effort required to support them as we evolve our APIs. These requirements should be included as part of any external API documentation.
+
+## Clients should conform to the HTTP Spec
+
+While the whole spec is important, it is worth calling out specific common cases where we expect clients to conform: 
+
+* Clients should follow HTTP-level redirects when a resource has moved to a new location.
+* Clients should obey any caching directives returned with a resource
```

Comment:
This is correct. There are no caching directives that require clients cache a resource, therefore you can "obey" caching directives by caching nothing.

Should I make this clearer in the doc?